ompi/mca/sharedfp/sm/sharedfp_sm_file_open.c

```diff
@@ -135,7 +135,6 @@ int mca_sharedfp_sm_file_open (struct ompi_communicator_t *comm,
         return OMPI_ERROR;
     }
 
-        free(sm_filename);
```

Comment:
you caught it, wow :-) No, technically its not, its a separate fix, somebody added it to the file (probably as part of the covertiy cleanup), but it is completely wrong, it actually makes this component seg fault. I can commit it separately if you want.